### PR TITLE
vimc-4226 add build to buildkite

### PR DESCRIPTION
**Pull request by hillalex**
This adds a buildkite pipeline, with a bit of re-organisation of files to allow for parallelisation and steps being run on different agents, as well as pulling out common env vars into a `common` file as per our pattern elsewhere.

### CODE REVIEW

File: scripts/run-build.sh [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+docker build -f build.dockerfile \
+    -t $BUILD_ENV_TAG \
+    --build-arg MONTAGU_GIT_ID=$GIT_SHA \
+    --build-arg MONTAGU_GIT_BRANCH=$GIT_BRANCH \
+    .
 
 # This is the path for teamcity agents. If running locally, pass in your own docker config location
 # i.e. /home/{user}/.docker/config.json
-docker_auth_path=${1:-/opt/teamcity-agent/.docker/config.json}
+docker_auth_path=${1:-/var/lib/buildkite-agent/.docker/config.json}
```
Comment (hillalex) [inline scripts/run-build.sh:15]: Note this has broken the TC build, but that will be removed in any case once this PR is approved.

**Comment by hillalex** [comment]
Nb I broke the TC build, so I've just deleted it already

File: dev/buildMinimal.dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM montagu-reverse-proxy-shared-build-env
+ARG MONTAGU_GIT_ID="UNKNOWN"
```
Comment (EmmaLRussell) [inline dev/buildMinimal.dockerfile:1]: Don't need this again on line 4?

File: dev/run-build-minimal.sh [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
-    --build-arg MONTAGU_GIT_ID=$git_id \
-    --build-arg MONTAGU_GIT_BRANCH=$git_branch \
+    --build-arg MONTAGU_GIT_ID=$GIT_SHA \
+    --build-arg MONTAGU_GIT_BRANCH=$GIT_BRANCH \
     .
 
 # This is the path for teamcity agents. If running locally, pass in your own docker config location
 # i.e. /home/{user}/.docker/config.json
-docker_auth_path=${1:-/opt/teamcity-agent/.docker/config.json}
+docker_auth_path=${1:-/var/lib/buildkite-agent/.docker/config.json}
```
Comment (EmmaLRussell) [inline dev/run-build-minimal.sh:14]: Could update the comment above this referencing teamcity

File: scripts/run-build.sh (hunk 1 above)
Comment (EmmaLRussell) [inline scripts/run-build.sh:15]: Another teamcity comment here

**Review by EmmaLRussell — COMMENTED** [review]
Looks good, like the emoji! :tada: 

I guess we can retire the TEAMCITY_VERSION env var in the shared build env dockerfile.

**Comment by hillalex** [comment]
Thanks, also noticed that the README was still referencing TC - all refs should be updated now!

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**